cmd/cassandra-writer/main.go

```diff
@@ -31,33 +29,35 @@ const (
 	svcName = "cassandra-writer"
 	sep     = ","
 
-	defNatsURL     = nats.DefaultURL
-	defLogLevel    = "error"
-	defPort        = "8180"
-	defCluster     = "127.0.0.1"
-	defKeyspace    = "mainflux"
-	defDBUsername  = ""
-	defDBPassword  = ""
-	defDBPort      = "9042"
-	defChanCfgPath = "/config/channels.toml"
-
-	envNatsURL     = "MF_NATS_URL"
-	envLogLevel    = "MF_CASSANDRA_WRITER_LOG_LEVEL"
-	envPort        = "MF_CASSANDRA_WRITER_PORT"
-	envCluster     = "MF_CASSANDRA_WRITER_DB_CLUSTER"
-	envKeyspace    = "MF_CASSANDRA_WRITER_DB_KEYSPACE"
-	envDBUsername  = "MF_CASSANDRA_WRITER_DB_USERNAME"
-	envDBPassword  = "MF_CASSANDRA_WRITER_DB_PASSWORD"
-	envDBPort      = "MF_CASSANDRA_WRITER_DB_PORT"
-	envChanCfgPath = "MF_CASSANDRA_WRITER_CHANNELS_CONFIG"
+	defNatsURL          = nats.DefaultURL
+	defLogLevel         = "error"
+	defPort             = "8180"
+	defCluster          = "127.0.0.1"
+	defKeyspace         = "mainflux"
+	defDBUsername       = ""
+	defDBPassword       = ""
+	defDBPort           = "9042"
+	defChannelsCfgPath  = "/config/channels.toml"
+	defSubtopicsCfgPath = "/config/subtopics.toml"
```

Comment:
I am not sure that we need to split this in 2 different configs. It would be better to have just one - like `subjects.toml` which would actually list the whole NATS subject (channel + subtopic)

Reply:
@drasko I proposed to @jonathandreyer to start by adding a new file. It's simplier and retro-compatible but if you think that it's ok to merge both in this PR I'm also ok

Reply:
@dusanb94 What do you think about that (rename file to `subjects.toml`)?

Reply:
My proposal for `subjects.toml` came from the fact that we can make Writers independent of the channels, and provide an arbitrary NATS subject to which they can subscribe and consume the data.

This proved as a beneficial option especially on the gateway where we had some applications that are not necessarily going through adapters, but are sending data directly on NATS, because they are internal to Mainflux Edge system (for example Influx Telegraf does that). Of course - these applications should send data in SenML format for now in order for Writers to parse it, but for now we create these applications (so we can control their format), and in the future we can thing on generalizing Writers.

In any case - I think that approach with NATS subjects is the most generic one, and then user can put any subject he wants - with channel only or with channel + subtopic, or something totally different that is meaningful to his application.

Reply:
@drasko Makes sense to me. Initially, I thought that knowing NATS topics is an unnecessary burden for the user, but the user needs to know topics to which messages are published anyways.

Reply:
@drasko @dusanb94 We merge both files into one file named `subjects.toml`? If it is ok for you, can we take this template of file?
```
[channels]
filter = ["*"]

[subtopics]
filter = ["*"]
```

Reply:
@jonathandreyer Yes, that looks fine.

Reply:
Do we need a separate table?

Why not just
```toml
subjects = [...]
```

Reply:
@drasko Technically, it is not in the same field we need to check. For channels, it is the `Channel` field and for subtopics, it is the `Subtopics` field.

Reply:
@jonathandreyer I am OK with the previous proposal also, because probably conf file parser would expect something like this. Please keep previous form then, and we'll se how it looks like in a corrected PR.

Reply:
@drasko What do you mean by `Please keep previous form then, and we'll se how it looks like in a corrected PR.`?

Reply:
I keep the older file `channels.toml` and create a new one `subject.toml`?

Reply:
I ment you can keep:
```toml
[subjects]
filter = ["*"]
```

All you need is an array of NATS subjects to which Writer will subscribe to.

Reply:
So, I don't merge the two configuration files? Actually, the PR is in this state without the merge of two configuration files.

Reply:
No, no merging. Rather implementation of this design idea - generic NATS subject list.

Reply:
Ok I understand your idea, but I have no idea how to proceed. Have you an idea what you would to have? Can you describe a example of what you would like to have?
When you say `generic NATS subject list`, you would like replace the actual mechanism (channels.toml or channels.toml+subtopics.toml) by something like:
```
subject = ["*.bedroom.temperature"]
```

Reply:
Either by:
```toml
subjects = ["*.bedroom.temperature", "something.else", ...]
```
Or by:
```toml
[subjects]
filter = ["*"]
```

But it boils down to the same thing - TOML file should provide a list of NATS subjects to which Writer will subscribe.

Reply:
Can we use `defSubjectsCfgPath  = "/config/subjects.toml"` and: 
```
[subjects]
channels=["*"]
subtopics=["*"]
```

Reply:
No need for `channels` and `subtopics` anymore - `subject` is made by concatenating `channel` and `subtopic` (example `subtopic = channels/<chan_id>/messages/a/b/c`)